### PR TITLE
Rename researchdelight hub to showcase

### PR DESCRIPTION
**Pull request by sgibson91**
- https://github.com/2i2c-org/infrastructure/issues/3279

### Things changed in this PR

- filenames that used researchdelight
- specific parts of config that references research delight
- sets up a redirect from researchdelight.2i2c.cloud to showcase.2i2c.cloud

### Things not changed in this PR

- Kubernetes annotations that still use researchdelight. This PR does not update terraform in any way. These annotations cannot be seen by a user anyway so should not be problematic.
- Team names that contain research delight. Awaiting new teams.
- Image names that contain research delight. Awaiting new image.

### What to do AFTER approval, but BEFORE merge

- [x] Update A/CNAME records in Namecheap for new name
- [x] Update URLs in GitHub OAuth app

### What to do AFTER merge

- [ ] delete the old `researchdelight` namespace in helm

### CODE REVIEW

File: config/clusters/2i2c-aws-us/cluster.yaml [hunk 1]
```diff
@@ -26,14 +26,14 @@ hubs:
       - daskhub-common.values.yaml
       - dask-staging.values.yaml
       - enc-dask-staging.secret.values.yaml
-  - name: researchdelight
-    display_name: "2i2c Research Delight"
-    domain: researchdelight.2i2c.cloud
+  - name: showcase
```
Comment (sgibson91) [inline config/clusters/2i2c-aws-us/cluster.yaml:29]: This change means that a new hub and helm namespace will be deployed. Hence the need to delete the old `researchdelight` namespace documented in the top comment.

**Review by consideRatio — APPROVED** [review]

**Comment by consideRatio** [comment]
I appreciate seeing a redirect setup as well! Looks great!

**Comment by sgibson91** [comment]
Deployment of the support chart failed with:

```
Running helm upgrade --install --create-namespace --namespace=support --wait support /home/runner/work/infrastructure/infrastructure/helm-charts/support --values=/tmp/tmppb3i_j17 --values=/home/runner/work/infrastructure/infrastructure/config/clusters/2i2c-aws-us/support.values.yaml --values=/tmp/tmp3fl8bzz9
Error: UPGRADE FAILED: failed to create resource: admission webhook "validate.nginx.ingress.kubernetes.io" denied the request: nginx.ingress.kubernetes.io/configuration-snippet annotation cannot be used. Snippet directives are disabled by the Ingress administrator
```

**Comment by sgibson91** [comment]
Wondering if this is relevant? https://github.com/kubernetes/ingress-nginx/issues/7837

**Pull request merged**